app/models/king.rb

```diff
@@ -1,18 +1,53 @@
 class King < Piece
-  # ensures king piece can only move 1 square in any direction
+
   def valid_move?(x,y)
     # checks critera for valid_move? that apply to all pieces and returns false if fails
     return false unless super(x,y)
     x_diff = x_diff(x)
     y_diff = y_diff(y)
     # returns true if moving only 1 square any direction
-    if (x_diff <= 1) && (y_diff <= 1)
-      return true
-    end
-    # returns false if not moving only 1 square any direction
+    return true if (x_diff <= 1) && (y_diff <= 1)
+
+    #returns true if the king can castle
+    return true if can_castle?(x,y)
+
+    # returns false if not moving only 1 square any direction and can't castle
     return false
   end
 
+  def can_castle?(x,y)
+    never_moved? && (castle_kingside?(x,y) || castle_queenside?(x,y))
+  end
+
+  #determines if king is attempting to castle on the kingside
+  def castle_kingside?(x,y)
+    x == 6 && y == y_position && castle_rook && \
+    castle_rook.x_position == 7 && castle_rook.never_moved? && \
+    castle_rook.path_blocked?(4, y_position) == false
+  end
+
+  #determines if king is attempting to castle on the queenside
+  def castle_queenside?(x,y)
+    x == 2 &&  y == y_position && castle_rook && \
+    castle_rook.x_position == 0 && castle_rook.never_moved? && \
+    castle_rook.path_blocked?(4, y_position) == false 
+  end
+
+  #determines if there is a rook to castle with
+  def castle_rook
+    @castle_rook = game.pieces.where(y_position: y_position, user_id: user_id, type: 'Rook').first 
```

Comment:
This method returns the first rook, not necessarily the correct rook.  This is causing some of your test failure.  Specifically, the first rook (`.first`), if neither have been moved, is always the queenside one, however your test for kingside castle fails because this method doesn't ever return the kingside rook.  You will want to either update this method to support choosing which rook you want, split it into two methods for each side, or refactor the logic into the other methods, based on what makes sense (to you, it's kinda subjective). 

👍